nuxeo-features/nuxeo-automation/nuxeo-automation-scripting/src/main/java/org/nuxeo/automation/scripting/internals/AutomationScriptingComponent.java

```diff
@@ -57,22 +68,52 @@ public void start(ComponentContext context) {
 
     @Override
     public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
-        if (contribution instanceof ScriptingOperationDescriptor) {
+        if (XP_OPERATION.equals(extensionPoint)) {
             registry.addContribution((ScriptingOperationDescriptor) contribution);
+        } else if (XP_CLASSFILTER.equals(extensionPoint)) {
+            registerClassFilter((ClassFilterDescriptor) contribution);
         } else {
             log.error("Unknown extension point " + extensionPoint);
         }
     }
 
     @Override
     public void unregisterContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
-        if (contribution instanceof ScriptingOperationDescriptor) {
+        if (XP_OPERATION.equals(extensionPoint)) {
             registry.removeContribution((ScriptingOperationDescriptor) contribution);
+        } else if (XP_CLASSFILTER.equals(extensionPoint)) {
+            unregisterClassFilter((ClassFilterDescriptor) contribution);
         } else {
             log.error("Unknown extension point " + extensionPoint);
         }
     }
 
+    protected void registerClassFilter(ClassFilterDescriptor desc) {
+        classFilterDescriptors.add(desc);
+        recomputeClassFilters();
+    }
+
+    protected void unregisterClassFilter(ClassFilterDescriptor desc) {
+        classFilterDescriptors.remove(desc);
+        recomputeClassFilters();
+    }
+
+    protected void recomputeClassFilters() {
+        Set<String> classNames = new HashSet<>();
```

Comment:
Its probably clearer to call this variable `allowedClassNames`, I had a long look at the if/else logic because `classNames.addAll(desc.allow)` is in both the if and else.  I understand now because its building a list of allowed class names.

Reply:
You're right. Updated.